core/src/main/java/io/questdb/griffin/engine/functions/json/JsonPathFunc.java

```diff
@@ -0,0 +1,153 @@
+    private final VarcharSupportingState copied = new VarcharSupportingState();
+    private final String functionName;
+    private final Function json;
+    private final int maxSize;
+    private final Function path;
+    private final DirectUtf8Sink pathSink;
+    private final boolean strict;
+
+    public JsonPathFunc(String functionName, Function json, Function path, DirectUtf8Sink pathSeq, int maxSize, boolean strict) {
+        this.functionName = functionName;
+        this.json = json;
+        this.path = path;
+        this.pathSink = pathSeq;
+        this.maxSize = maxSize;
+        this.strict = strict;
+    }
+
+    @Override
+    public void close() {
+        copied.close();
+        a.close();
+        b.close();
+        json.close();
+        path.close();
+        pathSink.close();
+    }
+
+    @Override
+    public Function getLeft() {
+        return json;
+    }
+
+    @Override
+    public Function getRight() {
+        return path;
+    }
+
+    @Override
+    public void getVarchar(Record rec, Utf8Sink utf8Sink) {
+        if (utf8Sink instanceof DirectUtf8Sink) {
+            copied.destSink = (DirectUtf8Sink) utf8Sink;
+            jsonPath(functionName, json.getVarcharA(rec), pathSink, copied, maxSize, strict);
+        } else {
+            // Extra intermediate copy from malloc'd memory to java memory required.
+            // TODO: Should the `utf8Sink` sink be cleared here?
+            //       Should I validate a minimum residual capacity?
+            //       At the moment, the `maxSize` passed in is the min of whatever is available and
+            //       what is the actual max.
+            // final int curtailedMaxSize = (int) Math.min(maxSize, utf8Sink.capacity());  /// utf8Sink `capacity` is missing!
+            final int curtailedMaxSize = maxSize;
+            if (copied.destSink == null) {
+                copied.destSink = new DirectUtf8Sink(curtailedMaxSize);
+            } else {
+                copied.destSink.clear();
+                copied.destSink.reserve(curtailedMaxSize);
+            }
+            jsonPath(functionName, json.getVarcharA(rec), pathSink, copied, curtailedMaxSize, strict);
+            utf8Sink.put(copied.destSink);
+        }
+    }
+
+    @Override
+    public @Nullable DirectUtf8Sink getVarcharA(Record rec) {
+        return jsonPath(functionName, json.getVarcharA(rec), pathSink, a, maxSize, strict);
+    }
+
+    @Override
+    public @Nullable DirectUtf8Sink getVarcharB(Record rec) {
+        return jsonPath(functionName, json.getVarcharB(rec), pathSink, b, maxSize, strict);
+    }
+
+    private static @Nullable DirectUtf8Sink jsonPath(
+            String functionName,
+            @Nullable Utf8Sequence json,
+            @Nullable DirectUtf8Sequence path,
+            VarcharSupportingState state,
+            int maxSize,
+            boolean strict
+    ) {
+        if (json == null || path == null) {
```

Comment:
path is a constant, it need not to be evaluated for each row. When path is null, result could be constant null function (done in factory)